### PR TITLE
adds logic and endpoint to accelerate control server fetch interval

### PR DESCRIPTION
**Pull request by James-Pickett**

### CODE REVIEW

File: ee/localserver/server.go [hunk 1]
```diff
@@ -129,8 +129,8 @@ func New(db *bbolt.DB, kolideServer string, opts ...LocalServerOption) (*localSe
 	// mux.Handle("/query", ls.requestQueryHandler())
 	// curl localhost:40978/scheduledquery --data '{"name":"pack:kolide_device_updaters:agentprocesses-all:snapshot"}'
 	// mux.Handle("/scheduledquery", ls.requestScheduledQueryHandler())
-	// curl localhost:40978/controlservicefetch
-	// mux.Handle("/controlservicefetch", ls.requestControlServiceFetch())
+	// curl localhost:40978/acceleratecontrol  --data '{"interval":"250ms", "duration":"1s"}'
```
[the remark this answers is not in the document]
Comment (James-Pickett) [reply ee/localserver/server.go:132]: cool, thanks! yes, the curl is just an example we use for local testing. The real thing would be in a krypto NaCL box from the JS bridge

File: ee/control/control.go [hunk 2]
```diff
@@ -110,6 +112,45 @@ func (cs *ControlService) Stop() {
 	}
 }
 
+func (cs *ControlService) AccelerateRequestInterval(interval, duration time.Duration) error {
+	// preform a fetch now
```
Comment (seejdev) [inline ee/control/control.go:116]:
```suggestion
	// perform a fetch now
```

File: cmd/launcher/control.go [hunk 3]
```diff
@@ -42,7 +42,7 @@ func createControlService(ctx context.Context, logger log.Logger, store types.Ge
 		control.WithRequestInterval(opts.ControlRequestInterval),
 		control.WithStore(store),
 	}
-	service := control.New(logger, ctx, client, controlOpts...)
+	service := control.New(logger, client, controlOpts...)
```
Comment (directionless) [inline cmd/launcher/control.go:45]: Thank you for pulling out `ctx` it seems wrong here

File: ee/control/control.go [hunk 4]
```diff
@@ -110,6 +112,45 @@ func (cs *ControlService) Stop() {
...
+	if err := cs.Fetch(); err != nil {
+		// if we got an error, log it and move on
+		level.Error(cs.logger).Log("msg", "failed to fetch data from control server. Not fatal, moving on", "err", err)
```
Comment (directionless) [inline ee/control/control.go:119]: Usually s/error/debug/

File: ee/control/control.go [hunk 5]
```diff
@@ -110,6 +112,45 @@ func (cs *ControlService) Stop() {
...
+	}
+
+	if interval <= 0 || duration <= 0 {
```
Comment (directionless) [inline ee/control/control.go:122]: I think we should define a minimum interval, I'd say 5 or 10s, and use that if it's lower

**Review by directionless — COMMENTED** [review]
I'm generally fine with this. I think parts will shuffle around as we bring in more interfaces and `kontext`

File: ee/control/control_test.go [hunk 6]
```diff
@@ -273,3 +277,101 @@ func TestControlServicePersistLastFetched(t *testing.T) {
...
+			accelerationInterval: 250 * time.Millisecond,
+			accelerationDuration: 1 * time.Second,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			// Overly verbose way to figure out how many fetches we expect to make it more human readable (hopefully)
```
Comment (RebeccaMahany) [inline ee/control/control_test.go:312]: This test is readable and makes sense 👍

**Review by RebeccaMahany — DISMISSED** [review]

File: ee/control/control.go (hunk 5 above)
Comment (James-Pickett) [reply ee/control/control.go]: done

File: ee/control/control.go (hunk 4 above)
Comment (James-Pickett) [reply ee/control/control.go]: done, nice use of sed =)

File: cmd/launcher/control.go (hunk 3 above)
Comment (James-Pickett) [reply cmd/launcher/control.go:45]: it wasn't being used

**Review by RebeccaMahany — APPROVED** [review]

**Review by seejdev — APPROVED** [review]

**Pull request merged**